pkgs/development/libraries/libusbsio/default.nix

```diff
@@ -0,0 +1,38 @@
+{ lib, stdenv, fetchzip, pkg-config, libusb, systemdMinimal, ... }:
+let
+  libSuffix = if stdenv.isDarwin then "dylib" else "so";
+  binDirPrefix = if stdenv.isDarwin then "osx_" else "linux_";
+in
+stdenv.mkDerivation {
+  pname = "libusbsio";
+  version = "2.1.11";
+
+  src = fetchzip {
+    url = "https://www.nxp.com/downloads/en/libraries/libusbsio-2.1.11-src.zip";
+    sha256 = "sha256-qgoeaGWTWdTk5XpJwoauckEQlqB9lp5x2+TN09vQttI=";
+  };
+
+  postPatch = ''
+    rm -r bin/*
+  '';
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [
+    libusb
+    systemdMinimal.dev # libudev
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    install -D bin/${binDirPrefix}$(uname -m)/libusbsio.${libSuffix} $out/lib/libusbsio.${libSuffix}
```

Comment:
I don't think we want to use uname -m here. Can't we get that from a nix variable or use a wildcard?

Reply:
I don't know but I would suspect that it could be a source for impurity.